src/OSPSuite.SimModelNative/src/FormulaChange.cpp

```diff
@@ -101,6 +110,14 @@ bool FormulaChange::PerformSwitchUpdate(double * y, double time)
 		return false; //same formula already set - no change
 
 	//really new formula - set new FORMULA
+	if (!_quantity->GetFormula())
+	{
+		set<int> usedIDs;
+		set<int> empty;
+		_newFormula->AppendUsedVariables(usedIDs, empty);
+		if (usedIDs.size() > 0)
+			switchJacobians = true;
+	}
```

Comment:
ok, [**almost correct** now](https://external-preview.redd.it/-fQgGV1pi6bK2fcxDCFX-GGy8ltk_Yzo5kv0wFhVI7s.jpg?width=960&crop=smart&auto=webp&s=fbe513ea103d32fa7a9c189b15a9cd239bfc5924) 🙂 

Here, we must append used variables ids from **both the old and the new formula**.
So it should be something like this:
```
...
_newFormula->AppendUsedVariables(usedIDs, empty);
if (_quantity->GetFormula())
   _quantity->GetFormula()->AppendUsedVariables(usedIDs, empty);
if (usedIDs.size() > 0)
...
```